lib/public_cocina_record.rb

```diff
@@ -3,27 +3,77 @@
 require 'http'
 
 class PublicCocinaRecord
-  attr_reader :druid, :purl_url
+  attr_reader :public_cocina_doc, :druid, :purl_url
 
-  def self.fetch(url)
-    response = HTTP.get(url)
-    response.body if response.status.ok?
+  def self.fetch(druid, purl_url: 'https://purl.stanford.edu')
+    response = HTTP.get("#{purl_url}/#{druid}.json")
+    new(druid, response.body, purl_url:) if response.status.ok?
   end
 
-  def initialize(druid, purl_url: 'https://purl.stanford.edu')
+  def initialize(druid, public_cocina, purl_url: 'https://purl.stanford.edu')
     @druid = druid
     @purl_url = purl_url
+    @public_cocina_doc = JSON.parse(public_cocina)
+  end
+
+  def public_cocina
+    @public_cocina ||= self.class.fetch("#{purl_url}/#{druid}.json")
+  end
+
+  def cocina_access
+    @cocina_access ||= Cocina::Models::DROAccess.new(public_cocina_doc['access'])
```

Comment:
The reason I didn't use the cocina structures here is that it means we need to keep in sync with the version of cocina models.   It's safer to just use the JSON in case the data doesn't match the structures.

Reply:
Would we want to use that as a warning that stuff needs remediation, and avoid indexing it entirely? I guess I didn't realize that things could get substantially out of sync. That seems like the problem that cocina-models is supposed to solve.

Reply:
Added a commit to remove all the cocina-models usage except for `TitleBuilder`, which seemed non-trivial, and one instance of constructing fixture data in a test.